### PR TITLE
fix(logout): notify on logout

### PR DESCRIPTION
**Pull request by mwegman**

### CODE REVIEW

File: src/spark-core.js [hunk 1]
```diff
@@ -143,7 +143,11 @@ var SparkCore = AmpersandState.extend({
       .catch(function logDeviceRemovalFailure(reason) {
         this.logger.warn(reason);
       })
-      .then(this.credentials.logout.bind(this.credentials));
+      .then(function cleanUpAndNotify() {
+        this.credentials.logout();
```
Comment (ianwremmel) [inline src/spark-core.js]: this function returns a promise, so i think you either want to trigger before calling credentials.logout, or cache its promise, trigger, then return its promise

Comment (mwegman) [inline src/spark-core.js]: Perhaps I'm wrong but I didn't think 'credentials.logout' returned a promise.

Comment (ianwremmel) [inline src/spark-core.js]: Ah, gotcha, that could be. looking a little deeper, `Credentials#logout()` redirects to the CI logout page, so all other logout work needs to come before that.

**Comment by ianwremmel** [comment]
Looks good, but let's hold off on merging until verifying it works as expected in the web client. Since indexedDB is async, I'm not sure it's guaranteed to clear all your keys (when the logout event fires) before being redirected.

File: src/spark-core.js [hunk 2]
```diff
@@ -143,7 +145,14 @@ var SparkCore = AmpersandState.extend({
       .catch(function logDeviceRemovalFailure(reason) {
         this.logger.warn(reason);
       })
-      .then(this.credentials.logout.bind(this.credentials));
+      .then(function clearEncryptStore() {
+        this.encryption.keystore.clear();
```
Comment (ianwremmel) [inline src/spark-core.js]: Does clear return a promise?

Comment (mwegman) [inline src/spark-core.js]: Clearing the keys return a promise but unfortunately that promise isn't propagated up by ampersand.

**Comment by ianwremmel** [comment]
LGTM

**Pull request merged**